app/src/main/java/com/example/instagramclone/fragments/profile_screen/compose/ProfileScreenComposable.kt

```diff
@@ -0,0 +1,205 @@
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.ExperimentalLifecycleComposeApi
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavHostController
+import coil.compose.AsyncImage
+import com.example.instagramclone.R.drawable
+import com.example.instagramclone.fragments.compose.components.BottomBar
+import com.example.instagramclone.fragments.profile_screen.ProfileScreenState
+import com.example.instagramclone.fragments.profile_screen.ProfileScreenState.Success
+import com.example.instagramclone.fragments.profile_screen.ProfileScreenViewModel
+import com.example.instagramclone.fragments.ui.theme.EtGrey2
+
+@OptIn(ExperimentalLifecycleComposeApi::class, ExperimentalAnimationApi::class)
+@Composable
+fun ProfileScreenComposable(
+    navController: NavHostController,
+    profileScreenViewModel: ProfileScreenViewModel = hiltViewModel()
+) {
+    val profileState by profileScreenViewModel.uiState.collectAsStateWithLifecycle()
+    when (profileState) {
+        is ProfileScreenState.Success -> {
+            AnimatedContent(profileState) {
+                Scaffold(
+                    topBar = {
+                        ProfileScreenTopBar()
+                    },
+                    bottomBar = {
+                        BottomBar(navHostController = navController)
+                    }
+                ) { paddingValues ->
+                    LazyVerticalGrid(
+                        modifier = Modifier
+                            .padding(paddingValues)
+                            .fillMaxSize(),
+                        columns = GridCells.Fixed(3)
+                    ) {
+                        item(span = {
+                            GridItemSpan(3)
+                        }) {
+                            ProfileDetailSection()
+                        }
+                        item(span = {
+                            GridItemSpan(3)
+                        }) {
+                            EditProfileButtons()
+                        }
+
+                        items((profileState as Success).savedPostList) { savedPostItem ->
+                            GridPostItem(savedPostItem)
+                        }
+                    }
+                }
+            }
+        }
+        else -> Unit
+    }
+}
+
+@Composable
+fun EditProfileButtons() {
+    Column() {
+        Row(
+            modifier = Modifier
+                .padding(8.dp)
+                .fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceAround
+        ) {
+            Button(
+                modifier = Modifier
+                    .height(36.dp)
+                    .width(334.dp)
+                    .clip(RoundedCornerShape(10.dp)),
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = MaterialTheme.colors.EtGrey2
```

Comment:
MaterialTheme has a colorPalette that has elements like `background`, `surface`, etc. Let's use those variables instead of direct color names.